Makefile

```diff
@@ -80,35 +90,34 @@ QEMU_PARAMS_DEBUG := -s &
 
 ISO_FILE := boot.iso
 
-.PHONY: iso
-iso: all
+$(ISO_FILE): all
 	@echo "GEN ISO" $(ISO_FILE)
-	@ grub-file --is-x86-multiboot $(TARGET) || { echo "Multiboot not supported"; exit 1; }
+	@ $(GRUB_FILE) --is-x86-multiboot $(TARGET) || { echo "Multiboot not supported"; exit 1; }
 	@ cp $(TARGET) grub/boot/
-	@ grub-mkimage --format i386-pc-eltorito -p /boot/grub -o grub/boot.img multiboot iso9660 biosdisk
-	@ xorriso -as mkisofs -U -b boot.img -no-emul-boot -boot-load-size 4 -boot-info-table -o $(ISO_FILE) grub 2>> /dev/null
+	@ $(GRUB_MKIMAGE) --format i386-pc-eltorito -p /boot/grub -o grub/boot.img $(GRUB_MODULES)
+	@ $(XORRISO) -as mkisofs -U -b boot.img -no-emul-boot -boot-load-size 4 -boot-info-table -o $(ISO_FILE) grub 2>> /dev/null
 
 .PHONY: boot
-boot: all
+boot: $(ISO_FILE)
 	@echo "QEMU START"
-	@qemu-system-x86_64 -cdrom $(ISO_FILE) $(QEMU_PARAMS)
+	@$(QEMU_BIN) -cdrom $(ISO_FILE) $(QEMU_PARAMS)
 
 .PHONY: boot_debug
-boot_debug: all iso
-	qemu-system-x86_64 -cdrom $(ISO_FILE) $(QEMU_PARAMS) $(QEMU_PARAMS_DEBUG)
+boot_debug: $(ISO_FILE)
+	$(QEMU_BIN) -cdrom $(ISO_FILE) $(QEMU_PARAMS) $(QEMU_PARAMS_DEBUG)
 
 .PHONY: run
-run: all
-	qemu-system-x86_64 -kernel $(TARGET) $(QEMU_PARAMS) $(QEMU_PARAMS_KERNEL)
+run: $(ISO_FILE)
+	$(QEMU_BIN) -kernel $(TARGET) $(QEMU_PARAMS) $(QEMU_PARAMS_KERNEL)
 
 .PHONY: debug
-debug: all
-	qemu-system-x86_64 -kernel $(TARGET) $(QEMU_PARAMS) $(QEMU_PARAMS_KERNEL) $(QEMU_PARAMS_DEBUG)
+debug: $(ISO_FILE)
```

Comment:
This should not depend on ISO_FILE, it needs kernel image only

Reply:
Alright. Fixing.